docs/content/integrations/dbt/reference.mdx

```diff
@@ -232,35 +223,71 @@ select *
 def orders():
     return ...
 ```
 
+This allows you to change asset keys within your dbt project without having to update the corresponding Dagster definitions.
+
+The <PyObject module="dagster_dbt" object="get_asset_key_for_source" /> method is used when a source has only one table. However, if a source contains multiple tables, like this example:
+
+```yaml
+sources:
+  - name: jaffle_shop
+    tables:
+      - name: raw_customers
+      - name: raw_orders
+      - name: raw_payments
+
+You can define a <PyObject object="multi_asset" decorator/> with keys from <PyObject module="dagster_dbt" object="get_asset_keys_by_output_name_for_source"/> instead:
+
+```python startafter=start_upstream_multi_asset endbefore=end_upstream_multi_asset file=/integrations/dbt/dbt.py dedent=4
+@multi_asset(
+    outs={
+        name: AssetOut(key=asset_key)
+        for name, asset_key in get_asset_keys_by_output_name_for_source(
+            [my_dbt_assets], "clients_data"
+        ).items()
+    }
+)
+def clients_data(context):
+    output_names = list(context.selected_output_names)
+    yield Output(value=..., output_name=output_names[0])
+    yield Output(value=..., output_name=output_names[1])
+```
+
 ### Downstream dependencies
 
-<Note>
-  Check out{" "}
-  <a href="/integrations/dbt/using-dbt-with-dagster/downstream-assets">
-    part four of the dbt + Dagster tutorial
-  </a>{" "}
-  to see this concept in context.
-</Note>
+Dagster allows you to define assets that are downstream of specific dbt models via <PyObject module="dagster_dbt" object="get_asset_key_for_model"/>. The below example defines `my_downstream_asset` as a downstream dependency of `my_dbt_model`:
 
-Dagster allows you to define assets that are downstream of specific dbt models. One property of dbt-based assets is that the external tool - in this case, dbt - handles storing each model in the database internally, rather than Dagster directly storing the tables that are updated.
+```python startafter=start_downstream_asset endbefore=end_downstream_asset file=/integrations/dbt/dbt.py dedent=4
+@asset(deps=[get_asset_key_for_model([my_dbt_assets], "my_dbt_model")])
+def my_downstream_asset():
+    ...
+```
 
-This means that there's a range of ways to load a dbt model as input to a Python function. For example, you might want to load the contents as a Pandas dataframe or into a PySpark session. You can specify this loading behavior on each downstream asset.
+Sometimes in the downstream asset, you may want direct access to the contents of the dbt model. Because dbt assets handle storing each model in the database internally, as opposed to Dagster directly storing the tables that are updated, there's a range of ways to load a dbt model as input to a Python function. For example, you might want to load the contents as a Pandas dataframe or into a PySpark session. You can specify this loading behavior on each downstream asset.
```

Comment:
In the spirit of deemphasizing IO managers, I would avoid language like "as opposed to Dagster directly storing the tables that are updated", because this is only the case when using Dagster with IO managers.

It would be better for the vibe to be something like "Dagster offers this cool IO manager thing that lets you separate business logic from IO, which is useful in some situations. You can use it if you want."

Relevant documentation for inspiration: https://docs.dagster.io/concepts/assets/software-defined-assets#defining-basic-managed-loading-dependencies

Reply:
Got it--there is also a whole section on "defining an IO manager": https://docs.dagster.io/integrations/dbt/reference#defining-an-io-manager. I deleted it in this PR